website/src/app/blog/how-dns-based-traffic-routing-works/readme.mdx

```diff
@@ -0,0 +1,372 @@
+equivalents across many organizations. The upside here is generally lower
+operational cost -- pay the businesses making the software to host it for you as
+opposed to hosting and managing it yourself, and reap the efficiency benefits in
+the form of cost reduction.
+
+But the downside with these public SaaS apps is that you can't use Split DNS to
+hide them anymore -- they aren't running on your infrastructure, so there simply
+aren't any internal zones and therefore no internal IPs to give to your
+workforce to connect to.
+
+As it turns out, there's a simple solution to this problem that's becoming more
+and more common across popular SaaS providers. Many public SaaS apps like
+GitHub, Slack, and Hubspot provide an IP allowlist feature that limits access to
+your tenant based on a range of allowed source addresses.
+
+Now, the studious reader will be quick to recognize the solution to our Split
+DNS problem: can't the organization simply tunnel the resolved IP addresses of
+these services through the VPN, just like they did before with internal
+services?
+
+And yes, that can work in some circumstances, but there's one minor problem
+standing in our way: **virtual hosting**.
+
+If the service uses
+[virtual hosting](https://en.wikipedia.org/wiki/Virtual_hosting), the IP address
+that the DNS query resolves to is shared among many different services, and not
+just the one the organization is interested in routing traffic for.
+
+<p className="text-sm text-center mx-auto italic">
+  Figure 5: Collateral damage can occur with Split DNS and some services
+</p>
+
+This means that an organization wishing to route DNS traffc _only_ for
+`github.com` might inadvertently route traffic for _other_ GitHub (and maybe
+even non-GitHub) services as well!
+
+#### NAT to the rescue?
+
+Ok, but what about another solution: can the organization simply configure its
+client devices to use the organization's DNS server to respond with internal IP
+addresses for the public services?
+
+That's much better -- it solves the problem of routing traffic for the wrong
+service, but it introduces a new problem: the organization now needs to maintain
+a list of all the public resources its workforce needs to access, then configure
+its DNS server to respond with a unique internal IP address for each one,
+finally it needs to be reasonably sure these IP addresses don't conflict with
+other networks available to the employee's machine, such as a home printer.
+
+<p className="text-sm text-center mx-auto italic">
+  Figure 6: Split DNS with NAT: A management nightmare
+</p>
+
+On top of the addressing headache, the organization needs to operate NAT routers
+that accept the traffic from the VPN clients across their workforce, translate
+the source addresses so they appear to come from one of the IPs in the
+allowlist, and finally forward the traffic on to the service in question. The IP
+addressing configuration and NAT router configuration would need to be kept in
```

Comment:
I'm not clear on what "The IP addressing configuration" means here

Reply:
Furthermore, the problem that an standard nat gateway doesn't solve is translating the destination address!

Reply:
Yeah I'll probably avoid going into the details of NAT here, and just mention how it's pain to manage all this stuff manually.